### DISCUSSION TITLE
New version

### DISCUSSION THREAD

**Discussion by gepbird** [samuel-lucas6/Kryptor#81] [General]
As [.NET 6 recently hit EOL](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#lifecycle) and this project already being on .NET 8, could you make a new release?

If I understand correctly, the only thing blocking the 4.1.1 release are the trimming issues with CommandLineUtils?
(or it may be already done, I just skimmed over the [commits since last release](https://github.com/samuel-lucas6/Kryptor/compare/v4.1.0...master) and the [roadmap](https://www.kryptor.co.uk/roadmap))

**Comment by samuel-lucas6** [comment]
Hi, thanks for using Kryptor.

Yes, a new release really needs to be done. I think the trimming can be resolved quite easily using a trick I found working on another program. Better erasure of secrets can be left for another release, but I need to update a dependency because there's a bug that affects Kryptor after switching to .NET 8 that doesn't appear during unit testing. I've fixed the library but am working on some other features that prevent me creating the package at the moment.

What bothers me about this type of program is that I'm manually testing it and not testing it properly cross-platform. Even trivial changes feel rather dangerous given what it's doing. Manual builds/signatures and sorting out packages also isn't very smooth. These things combined with other projects and IRL activities have meant I haven't done much work on this project for a while. Most of the development was done whilst I was at uni during holidays and when I had fewer projects.

I'll hopefully get the next version of the library out this weekend. However, Christmas is coming up so I'm not sure I'm going to be able to get Kryptor updated until the new year now.

**Reply by samuel-lucas6** [reply]
To update you on this, I was planning on doing a release the weekend just gone, but there's an issue with the libsodium NuGet package that prevents single-file publishing on macOS (x64 and ARM64). This is an upstream issue, so I either have to wait for it to be fixed or just do a release that isn't single-file for macOS. I'm thinking of giving it one or two weeks to see if it gets fixed, and if not, I'll just do a release. I doubt it'll get fixed that quickly to be honest due to Frank's other commitments/not having a .NET background.

**Reply by gepbird** [reply]
Thanks for the update! Waiting 2 weeks for upstream fix seems to be a reasonable decision

**Comment by gepbird** [comment]
Thanks for making a release!
